torch/csrc/jit/codegen/cuda/executor_utils.cpp

```diff
@@ -672,6 +689,42 @@ NvrtcFunction nvrtcCompile(
 #ifndef NDEBUG
   args.push_back("-lineinfo");
 #endif
+  args.push_back("-G");
+
+  // keeping this outside the loop for lifetime;
+  std::string max_register_usage = "--maxrregcount=";
+  // TODO: double check the math with occupancy header
+  if (opt_block_size.has_value()) {
+    // int num_partition;
+    // cudaOccSubPartitionsPerMultiprocessor(&num_partition, prop);
+    // status = cudaOccRegAllocationGranularity(
+    //     &allocationGranularity,
+    //     properties);
+    int num_partition = 4;
+    int reg_allocation_granularity = 256;
```

Comment:
Is this because the granularity per thread is 8?

Reply:
Yes.

In the occupancy header, they were associating the granularity per warp as 256. Since we always have warp size 32, we can put it 8 per thread as well.